probe-rs/src/session.rs

```diff
@@ -887,6 +896,19 @@ impl Permissions {
             Err(MissingPermissions("erase_all".into()))
         }
     }
+
+    /// Skip disabling debug peripherals when dropping the [Session]
+    #[must_use]
+    pub fn allow_dirty_detach(self) -> Self {
```

Comment:
I don't think the permissions are the correct place to do this semantically.
Maybe it's time to introduce proper `Options` struct to pass around? @Tiwalun any oppinions?